### PR TITLE
convert website from hugo to gatsby

### LINKED ISSUE #1351: [website] consider gatsby instead of hugo
## Change netlifycms.org website (website folder in this repo) to use Gatsby instead of Hugo

Seems like there could be benefits to using Gatsby instead of Hugo since netlify cms is React based and gatsby has a growing presence in the react static site realm. 

Reasons to move
- No need for all the gulp set up (remove a lot of deps)
- Would remove jquery plugins in favor of react components
- React templating benefits should be obvious here

I'd love to take a crack at this if there's buy-in. I'd just try to recreate the existing site 1:1

## Edit
I've got most of this done here
https://github.com/zebapy/netlify-cms/tree/gatsby-site/website

### PR DESCRIPTION
**Pull request by zebapy**
**- Summary**
Converts the entire docs site to Gatsby

closes #1351 

**- Test plan**
- [x] browser check ie11, edge, chrome, safari, firefox
- [x] test on latest ios (safari) and android (chrome)
- [x] run lighthouse performance test

## changes
- doc page titles are now in tab title
- headers in docs now have anchor links shown on hover like github
- ~~no longer js for adding table of content links in nav~~ had to revert to this (https://github.com/netlify/netlify-cms/pull/1369/commits/43b5b36dd521988d6fe2a00e96930d51c17fac8e) until https://github.com/gatsbyjs/gatsby/issues/5436 is closed  
- no more jquery :tada:
- video embed is self contained see see components/video-embed.js
- event widget is now self contained. see components/events-widget.js
- mobile menu logic is self contained. see components/mobile-nav.js
- docsearch initialization is mostly self contained. see components/doc-search.js. Docsearch script is still in layout file.
- page changes are instant. yay client side!
- sticky header uses react-headroom
- gitter chat is react-sidecar
- removed h1 headings from markdown files since title is in frontmatter and otherwise it becomes first heading in table of contents.
- removed docs link in sidebar docs nav. Do we really need this?

## remaining todos
- [x] docs nav style issues
- [x] remove sticky header on mobile
- [x] table of contents under nav item issues https://github.com/gatsbyjs/gatsby/issues/5436
- [x] event date local is wrong?
- [x] scroll to anchor animation of table of content links?
- [x] homepage cta spacing problem
- [x] need to watch postcss imports for changes - right now gatsby only watches the `main.css` file and rebuilds on changes to that one but it's not watching the imported files. ~~I'd recommend just using Sass to handle styles since it comes with everything that the postcss plugins are doing and would solve this but it may be unlikely the team wants that change.~~ EDIT: will just import css into component files
- [x] put root fields back in notifications.yml and updates.yml so cms can edit them

### from review 1
- [x] distance between icon and placeholder text in search input should be increased as in current site
- [x] width of code blocks should extend beyond the right margin as in current site
- [x] some inline code isn't parsing - see the bottom of this paragraph
- [x] the GitHub stars button in the header isn't displaying properly / loading issues

### review 2
- [x] Sticky header isn't sticking
- [x] inline code is styled same as blocks, should match styling currently used (light gray background)
- [x] _internal/pagination.html shown underneath list of blog posts

---

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by zebapy** [comment]
@erquhart build failed because public dir for the site should be `website/public`. Should I look into if I can change it on gatsby or can you change it?

**Comment by erquhart** [comment]
We can set the build directory via environment variable once this is merged and released: https://github.com/gatsbyjs/gatsby/pull/4756

Until then, we'll have to rename/move the directory in the npm `build` task.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 1102f9226a35e3a71ea26e0f4b3d4f4f568bf089

https://deploy-preview-1369--netlify-cms-www.netlify.com

**Comment by erquhart** [comment]
Note: Fetch is supported by all of the browsers that Netlify CMS supports.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by zebapy** [comment]
Oh ok so don't care about ie11 support? I'd be happy to revert the revert.

**Comment by erquhart** [comment]
Right, no IE support whatsoever 🤘🏼

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by zebapy** [comment]
I'll be trying to get back to this next week when I have some time off

**Comment by zebapy** [comment]
@erquhart I think I'm all set if you want to finish any review

**Comment by erquhart** [comment]
@zebapy so sorry for the delay, stuff got crazy there. Reviewing right now.

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-6
```
Comment (erquhart) [inline .nvmrc]: I feel like there was discussion of this change, but I can't recall. Can you?

Comment (zebapy) [reply .nvmrc]: Yup it's because `async` wasn't available so you recommended updating node here

**Comment by zebapy** [comment]
@erquhart could you provide screenshots or browser info about the GItHub stars? It looks ok to me in latest chrome besides it's slightly delayed load.

**Comment by verythorough** [comment]
I noticed some weirdness with some parts of the site taking a long time to load. For example, the stars button @erquhart mentions above was broken for me, too, until I'd had the site open for about 30 sec to a minute.  

It took a similar amount of time for docs pages to load. I could open the Intro doc right away, but when I clicked any links in the sidebar menu, the menu would change, but the page content would not. I clicked through several links for about 30 seconds or more, and then things started working as expected. Is there some kind of lazy loading going on here?

**Comment by zebapy** [comment]
@verythorough Content should be precached when changing pages https://www.gatsbyjs.org/docs/prpl-pattern/. I'm not able to reproduce general slowness. I do get the delayed/not loading GH buttons sometimes but I can look at that. Can you provide some sort of screencast?

**Comment by verythorough** [comment]
@zebapy I would have loved to include a screencast, but once it started working, it stayed that way. I though it might be a CDN caching thing, but I can't find a way to reproduce. Even the stars button pops in almost immediately no matter where/how I try (even in past deploys & new sites from forked repos). 
/shrug

**Comment by zebapy** [comment]
> distance between icon and placeholder text in search input should be increased as in current site

I have no idea what caused this but I added some styles to correct it. https://github.com/zebapy/netlify-cms/commit/d8bd2e75fe2d828d3037300880745f84efb9679c

> width of code blocks should extend beyond the right margin as in current site

fixed by https://github.com/zebapy/netlify-cms/commit/11f3d0539154bceae3cd0d411c27717f096d9492

> some inline code isn't parsing - see the bottom of this paragraph

Unfortunately this looks like a bug with transformer remark https://github.com/gatsbyjs/gatsby/issues/5417
I changed this paragraph emphasis to use asterisks and searched for others similar but couldn't find any. 

https://github.com/zebapy/netlify-cms/commit/0f6bb0cfafc1b9d379bbbd6fb322f8868659af4b

> the GitHub stars button in the header isn't displaying properly

Is this the same problem as https://github.com/netlify/netlify-cms/pull/1369#issuecomment-395614936 ?

**Comment by zebapy** [comment]
I started blog functionality in https://github.com/zebapy/netlify-cms/compare/gatsby-site...zebapy:gatsby-site-blog. Let me know if I should just merge this into `gatsby-site` branch

**Comment by erquhart** [comment]
I was just about to bring that up - you can just add it to this PR. We'll be merging the Hugo blog in first (like today), so a rebase will be in order after that. Thanks for staying on top of this, looking forward to getting it merged.

**Comment by zebapy** [comment]
@erquhart Ok I think I addressed your issues in your first review and added the basic blog set up. The one thing that's missing is pagination but I could add that later.

**Comment by erquhart** [comment]
@zebapy awesome!! Reviewing now.

**Comment by erquhart** [comment]
@zebapy looks good! Any thoughts on the sticky header?

**Comment by zebapy** [comment]
Sorry for delay. I should have that fixed this week.

**Comment by verythorough** [comment]
Hey, I just took another peek, and noticed that the widgets on the [Widgets doc](https://deploy-preview-1369--netlify-cms-www.netlify.com/docs/widgets/#default-widgets) aren't individually linkable like they are in the current setup (e.g. https://www.netlifycms.org/docs/widgets/#relation).

Thanks again for all your hard work!

**Comment by erquhart** [comment]
Also not seeing the mobile nav on iPhone, need to confirm on desktop.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by zebapy** [comment]
@verythorough the widgets should show now when url contains the hash.

@erquhart I need to fix that, it's hidden behind sticky header. Would be great to rethink how the sticky header functions so react-headroom could still be used (maybe just allow it on mobile). The `.hero:before` spacers are awkward.

**Comment by zebapy** [comment]
@verythorough I see you commented but deleted it? Not sure

**Comment by erquhart** [comment]
@zebapy definitely agreed on merging before any further improvements, will do soon.

**Comment by verythorough** [comment]
@zebapy I don't remember... I know it's been several days since I looked at this PR. I think the last thing I did was add emoji to https://github.com/netlify/netlify-cms/pull/1369#issuecomment-401501045 and switch my review to approved.

**Comment by zebapy** [comment]
anything you need from me on this?

**Comment by erquhart** [comment]
@zebapy nope, I'll get this merged up soon, just been head down in the monorepo split.

**Pull request merged**